src/Cake.Warp/ModuleInitializer.cs

```diff
@@ -0,0 +1,69 @@
+namespace Cake.Warp
+{
+    using System;
+    using System.IO;
+    using Cake.Warp.Common;
+    using Resourcer;
+
+    /// <summary>
+    /// Module class responible for the logic needed
+    /// to run when the library is loaded.
+    /// </summary>
+    public static class ModuleInitializer
+    {
+        /// <summary>
+        /// The actual method that will be called during
+        /// the loading of the addin library (by Module.Fody IL weaving).
+        /// </summary>
+        /// <notes>
+        ///   While this method is public, it should never be
+        ///   called manually. It should only be called
+        ///   using the existing IL weaving.
+        /// </notes>
+        public static void Initialize()
+        {
+            // TODO
```

Comment:
What is there left to do here?

Reply:
I completely forgot to add context to that todo statement, there is still something that needs to be done though.
I need to add an early check if the file exist, in which case we can skip the extraction of the binary file.

Another thing I forgot, is to investigate if the executing bit needs to be set for the binary extracted on linux and OSX. (I do believe that would be needed)